test/unit/fact_parser_test.rb

```diff
@@ -121,6 +121,88 @@ class FactParserTest < ActiveSupport::TestCase
     assert result[:bridge]
   end
 
+  test "#suggested_primary_interface detects primary interface using DNS" do
+    host = FactoryGirl.build(:host)
```

Comment:
A context block with a setup block for host & ```parser = get_parser``` would DRY this up a bit. 

Does the stubbed hash have to be ```.with_indifferent_access``` for some reason?